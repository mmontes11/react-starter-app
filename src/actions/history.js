import * as types from "constants/actionTypes/history";

export const setHistory = (squareIndex, symbol, step) => ({ type: types.SET_HISTORY, squareIndex, symbol, step });
