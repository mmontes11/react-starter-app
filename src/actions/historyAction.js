import * as types from "../constants/historyActionTypes";

export const setHistory = (squareIndex, symbol, step) => dispatch => {
  dispatch({ type: types.SET_HISTORY, squareIndex, symbol, step });
};
