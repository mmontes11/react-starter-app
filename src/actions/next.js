import * as types from "../constants/actionTypes/next";

export const nextPlayer = () => (dispatch, getState) => {
  dispatch({ type: types.NEXT_PLAYER, step: getState().step });
};
