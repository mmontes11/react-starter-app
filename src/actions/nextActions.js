import * as types from "../constants/nextActionTypes";

export const nextPlayer = () => (dispatch, getState) => {
  dispatch({ type: types.NEXT_PLAYER, step: getState().step });
};
