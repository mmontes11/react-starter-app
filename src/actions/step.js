import * as types from "../constants/actionTypes/step";

export const stepForward = () => ({ type: types.STEP_FORWARD });

export const jumpToStep = step => ({ type: types.JUMP_TO_STEP, step });
