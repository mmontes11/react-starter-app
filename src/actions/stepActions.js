import * as types from "../constants/stepActionTypes";

export const stepForward = () => ({ type: types.STEP_FORWARD });

export const jumpToStep = step => ({ type: types.JUMP_TO_STEP, step });
