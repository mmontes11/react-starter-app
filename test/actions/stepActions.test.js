import expect from "expect";
import { stepForward, jumpToStep } from "../../src/actions/stepActions";

describe("Step actions", () => {
  describe("Step forward", () => {
    it("Creates a step forward action", () => {
      const action = stepForward();
      expect(action).toBeAnAction();
      expect(action).toMatchSnapshot();
    });
  });
  describe("Jump to step", () => {
    it("Creates a jump to step action", () => {
      const action = jumpToStep(1);
      expect(action).toBeAnAction();
      expect(action).toMatchSnapshot();
    });
  });
});
