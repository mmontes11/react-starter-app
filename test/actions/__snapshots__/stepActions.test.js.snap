// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step actions Jump to step Creates a jump to step action 1`] = `
Object {
  "step": 1,
  "type": "JUMP_TO_STEP",
}
`;

exports[`Step actions Step forward Creates a step forward action 1`] = `
Object {
  "type": "STEP_FORWARD",
}
`;
